### PR TITLE
 Added user/password MQTT authentication and possibility to send "offline" messages too.

### PR DESCRIPTION
**Pull request by pavel-gololobov**

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -2,6 +2,10 @@
     <string name="app_name">Presence Publisher</string>
     <string name="host_title">MQTT Host</string>
     <string name="host_summary">IP address or hostname.%nCurrent value: %s</string>
+    <string name="login_title">MQTT Login</string>
+    <string name="login_summary">User Login.%nCurrent value: %s</string>
+    <string name="password_title">MQTT Password</string>
+    <string name="password_summary">User Password.%nCurrent value: %s</string>
```
Comment (ostrya) [inline app/src/main/res/values/strings.xml]: I'd prefer the password not being visible. Maybe just remove the second line with "current value"?

File: app/src/main/res/values-de/strings.xml [hunk 2]
```diff
@@ -45,4 +45,10 @@
     <string name="client_cert_summary">Das Client-Zertifikat, mit dem sich das Gerät authentifizieren soll.%nAktueller Wert: %s</string>
     <string name="last_ping_title">Letzter Erfolg</string>
     <string name="next_ping_title">Geplant für</string>
+    <string name="login_title">MQTT Anmeldung</string>
+    <string name="login_summary">Anmeldung.%nAktueller Wert: %s</string>
+    <string name="password_title">MQTT Passwort</string>
+    <string name="password_summary">Passwort.%nAktueller Wert: %s</string>
+    <string name="offlineping_title">Offline-Pings senden</string>
+    <string name="offlineping_summary">Senden Sie Pings, wenn WLAN nicht verbunden ist</string>
```
Comment (ostrya) [inline app/src/main/res/values-de/strings.xml]:
```suggestion
    <string name="offlineping_summary">Pings senden, wenn Ziel-WLAN nicht verbunden ist</string>
```

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -44,4 +48,6 @@
     <string name="client_cert_summary">Client certificate to authenticate this device.%nCurrent value: %s</string>
   <string name="last_ping_title">Last success</string>
   <string name="next_ping_title">Scheduled for</string>
+    <string name="offlineping_title">Send Offline Pings</string>
+    <string name="offlineping_summary">Send pings when WiFi disconnected</string>
```
Comment (ostrya) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="offlineping_summary">Send pings when target WiFi not connected</string>
```

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (pavel-gololobov) [reply app/src/main/res/values/strings.xml]: You're right. It will be a good solution.

**Comment by ostrya** [comment]
>
> I replace the password with asterisks. Maybe it is better to completely remove this line?

I guess it's fine like that for now. For now, there is no way to distinguish a set password from an empty one, but that would also be the case when deleting that line.

**Comment by pavel-gololobov** [comment]
Done.

**Pull request merged**